client/features/output-visualization-page/OutputVisualizationPage.tsx

```diff
@@ -1,21 +1,26 @@
 import React from 'react';
 import TreeVisualizationBox from './components/TreeVisualizationBox';
 import useStore from '@/client/stores/nodeMindMapStore';
+import { useMutation } from '@tanstack/react-query';
+import axios from 'axios';
 
 const OutputVisualizationPage = () => {
-  const saveState = () => {
-    const { nodes, edges } = useStore.getState(); // Access Zustand store state directly
-    console.log('Saving state:', { nodes, edges });
-    // Implement the save logic here, e.g., sending to a server or saving to localStorage
-  };
+  const saveFlowStateMutation = useMutation({
+    mutationFn: () => {
+      const { nodes, edges, treeId } = useStore.getState();
+      console.log('saving nodes to tree', nodes);
+      return axios.post(`/api/trees/${treeId}`, { nodes, edges });
```

Comment:
@ChTiSh This is just like in our client/server demo from a couple of months ago. We don't want Axios to reference the local folder - for one thing, when the app is deployed, the user won't have this folder - but we want the client to make its request to the server over the internet.

We should have something like an environment variable for SERVER_BASE_URL and in the Development environment this is set to "http://localhost:3000" for example. That variable can be inserted at the beginning of your resource string on line 12.